api/src/vtable.rs

```diff
@@ -0,0 +1,29 @@
+use crate::common::util::FStr;
+use fapi_common::vtable::HostVTable;
+
+static mut VTABLE: Option<HostVTable> = None;
+
+/// Intializes the vtable from the given string.
+///
+/// # Safety
+/// This function may not be called from multiple threads
+/// at once.
+pub fn init_vtable(s: FStr) -> Result<(), serde_json::Error> {
```

Comment:
Could we enforce this by having a static lock?

Reply:
Cost will likely be significant, since this is hot code. This is an internal API, so there's not too much need for enforcing safety, as long as we're careful. For now, though, I can switch this to a safe implementation.

Reply:
If the function has safety implications for calling, wouldn't it be better to make it an `unsafe fn` rather than wrap that up with a note in the docs?